static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -0,0 +1,436 @@
+
+  const viewsToTabs = views.map(
+    ({id, name, query: viewQuery, querySort: viewQuerySort}, index): Tab => {
+      return {
+        id: id ?? undefined,
+        key: `view-${index}`,
+        label: name,
+        query: viewQuery,
+        querySort: viewQuerySort,
+        queryCount: queryCounts[viewQuery]?.count ?? undefined,
+        to: normalizeUrl({
+          query: {
+            ...queryParams,
+            query: viewQuery,
+            sort: viewQuerySort,
+            ...(id ? {viewId: id} : {}),
+          },
+          pathname: `/organizations/${organization.slug}/issues/`,
+        }),
+      };
+    }
+  );
+
+  let initialTabKey = viewsToTabs[0].key;
+  if (defaultView === 'temporary-tab') {
+    initialTabKey = 'temporary-tab';
+  } else if (defaultView) {
+    initialTabKey =
+      viewsToTabs.find(tab => tab.id === defaultView)?.key ?? viewsToTabs[0].key;
+  }
+
+  const [draggableTabs, setDraggableTabs] = useState<Tab[]>(viewsToTabs);
+  const [selectedTabKey, setSelectedTabKey] = useState<string>(initialTabKey);
+  const [tempTab, setTempTab] = useState<Tab | undefined>(
+    initialTabKey === 'temporary-tab'
+      ? {
+          key: 'temporary-tab',
+          label: 'Unsaved',
+          query,
+          querySort: sortParam,
+          to: normalizeUrl({
+            query: {
+              ...queryParams,
+              query,
+              sort: sortParam,
+            },
+            pathname: `/organizations/${organization.slug}/issues/`,
+          }),
+        }
+      : undefined
+  );
+
+  const {mutate: updateViews} = useUpdateGroupSearchViews();
+
+  const debounceUpdateViews = debounce((newTabs: Tab[]) => {
+    if (newTabs) {
+      updateViews({
+        orgSlug: organization.slug,
+        groupSearchViews: newTabs.map(tab => ({
+          id: tab.id,
+          name: tab.label,
+          query: tab.query,
+          querySort: tab.querySort,
+        })),
+      });
+    }
+  }, 1000);
+
+  // Update URL with view's query, sort, and id(?) upon initial render. Ideally
+  // we can remove this after overview.tsx gets overhauled.
+  useEffect(() => {
+    router.replace(
+      normalizeUrl({
+        query: {
+          ...queryParams,
+          query: query,
+          sort: sort,
+          ...(draggableTabs[0]?.id ? {viewId: draggableTabs[0].id} : {}),
+        },
+        pathname: `/organizations/${organization.slug}/issues/`,
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // Update local tabs when new views are received from mutation request
+  useEffect(() => {
+    setDraggableTabs(
+      draggableTabs.map(tab => {
+        if (!tab.id) {
+          views.forEach(view => {
+            if (
+              tab.query === view.query &&
+              tab.querySort === view.querySort &&
+              tab.label === view.name
+            ) {
+              tab.id = view.id;
+              tab.to = normalizeUrl({
```

Comment:
I'm seeing this a lot. It might be better if the tabs had `query` and `sort` instead of `to`. Then `DraggableTabBar` could generate the link.